src/Avalonia.Controls/Primitives/PopupPositioning/CustomPopupPlacementCallback.cs

```diff
@@ -0,0 +1,6 @@
+namespace Avalonia.Controls.Primitives.PopupPositioning;
+
+/// <summary>
+/// Represents a method that provides custom positioning for a <see cref="Popup"/> control.
+/// </summary>
+public delegate void CustomPopupPlacementCallback(CustomPopupPlacement parameters);
```

Comment:
Why use this pattern rather than an attached event?

Reply:
Oh wow, ok looks like this comes from WPF. Seems a bit strange not to used an attached event.